src/OpenSkill/Datatable/Providers/QueryBuilderProvider.php

```diff
@@ -0,0 +1,251 @@
+        $this->compileQuery($this->columnConfiguration);
+
+        // sort
+        $this->sortQuery();
+
+        // limit
+        $this->queryBeforeLimits = clone $this->query;
+        $this->limitQuery();
+
+        // original # of items
+        $filteredItems = $this->originalQuery->count();
+
+        // # of items in filtered & ordered dataset
+        $dataCount = $this->queryBeforeLimits->count();
+
+        // the data for the response
+        $columns = $this->compileColumnNames();
+        $response = new Collection($this->query->get($columns));
+
+        // slice the result into the right size
+        return new ResponseData(
+            $response,
+            $filteredItems,
+            $dataCount
+        );
+    }
+
+    /**
+     * Will compile the collection into the final collection where operations like search and order can be applied.
+     *
+     * @param QueryBuilder $query
+     * @param ColumnConfiguration[] $columnConfiguration
+     * @return QueryBuilder
+     * @throws DatatableException
+     */
+    private function compileQuery(array $columnConfiguration)
+    {
+        if ($this->queryConfiguration->isGlobalSearch()) {
+            $this->compileGlobalQuery($columnConfiguration);
+        } elseif ($this->queryConfiguration->isColumnSearch()) {
+            $this->compileColumnQuery($columnConfiguration);
+        }
+    }
+
+    /**
+     * When a global (single) search has been done against data in the datatable.
+     *
+     * @param array $columnConfiguration
+     * @return QueryBuilder
+     * @throws DatatableException
+     */
+    private function compileGlobalQuery(array $columnConfiguration)
+    {
+        foreach ($columnConfiguration as $i => $col) {
+            $this->createQueryForColumn($col, $this->queryConfiguration->searchValue());
+        }
+    }
+
+    /**
+     * When a global query is being performed (ie, a query against a single column)
+     *
+     * @param ColumnConfiguration[] $columnConfiguration
+     * @return QueryBuilder
+     * @throws DatatableException
+     */
+    private function compileColumnQuery(array $columnConfiguration)
+    {
+        $searchColumns = $this->queryConfiguration->searchColumns();
+
+        foreach ($searchColumns as $i => $col) {
+            $column = $this->getColumnFromName($columnConfiguration, $col->columnName());
+
+            if (!isset($column))
+                continue;
+
+            $this->createQueryForColumn($column, $col->searchValue());
+        }
+    }
+
+    private function createQueryForColumn(ColumnConfiguration $column, $searchValue)
+    {
+        $searchType = $column->getSearch();
+
+        if ($searchType == DefaultSearchable::NONE()) {
+            // Don't do anything, this is not a searchable field
+            return $this->query;
+        } elseif ($searchType == DefaultSearchable::NORMAL()) {
+            $this->query->orWhere($column->getName(), 'LIKE', '%' . $searchValue . '%');
```

Comment:
I remember having problems with postgresql users that wanted to use `iLIKE` in their queries.

Should we consider that or providing the option to override any column query?